expertsystem/io/yaml/_build.py

```diff
@@ -15,8 +41,206 @@
 from . import validation
 
 
-def build_particle_collection(definition: dict) -> ParticleCollection:
-    validation.particle_list(definition)
+def build_amplitude_model(definition: dict) -> AmplitudeModel:
+    validation.amplitude_model(definition)
+    particles = build_particle_collection(definition, validate=False)
+    parameters = build_fit_parameters(definition["Parameters"])
+    kinematics = build_kinematics(definition["Kinematics"], particles)
+    dynamics = build_particle_dynamics(
+        definition["Dynamics"], particles, parameters
+    )
+    intensity = build_intensity(definition["Intensity"], particles, parameters)
+    return AmplitudeModel(
+        particles=particles,
+        kinematics=kinematics,
+        parameters=parameters,
+        intensity=intensity,
+        dynamics=dynamics,
+    )
+
+
+def build_fit_parameters(definition: List[dict]) -> FitParameters:
```

Comment:
Did you leave all of those functions, which "build" a portion of the full AmplitudeModel public on purpose?

Reply:
Yes it's used here
https://github.com/ComPWA/expertsystem/blob/ef358578b7583a79d1fe05110fd4ea727f0a0595/expertsystem/io/yaml/_build.py#L47

Or do you mean, why is it public?

Reply:
yes why is it public, and all of the functions except `build_amplitude_model`

Reply:
Ok c442ef3c ff8f6ba5 40dc58a0 makes the implementations private. We should think about merging these dumper modules though because of the many duplicate lines. For a follow-up issue.